plenum/server/pool_manager.py

```diff
@@ -272,6 +272,18 @@ def nodeHaChanged(self, txn):
         # TODO: Check if new HA is same as old HA and only update if
         # new HA is different.
         if nodeName == self.name:
+            # Update itself in node registry if needed
+            (ip, port) = self.node.nodeReg[nodeName]
+            if ip != txn[DATA][NODE_IP] or port != txn[DATA][NODE_PORT]:
```

Comment:
What if txn conatins IP only (or PORT only)?

Reply:
Bad things will happen in this case... Additional checks should be added.